File: DailyChallenges/CPP/LeetCode_2244.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> cnt;
        for (int t: tasks) 
            ++cnt[t];
        int ans = 0;
        for (auto [_, c]: cnt){
            if (c == 1) return -1;
            int d = c/3, r = c%3;
            if (r==0)
                ans += d;
            else // r == 1 or 2
                ans += d+1;
        }
        return ans;
    }
};

File: DailyChallenges/Scala/LeetCode_2244.scala
object Solution {
    def minimumRounds(tasks: Array[Int]): Int = {
        tasks.groupMapReduce(identity)(_ => 1)(_ + _).foldLeft(0)((c,p)=> {
            if (p._2 == 1) return -1
            c + (if (p._2 % 3 == 0) (p._2 / 3) else (p._2 / 3 + 1))
        })  
    }
}
